src/components/SemanticViewer.jsx

```diff
@@ -256,31 +257,42 @@ class SemanticViewer extends Component {
 
 	onClickDataset = async (semantic) => {
 		this.setState({
-			selectedDataset: semantic,
+			selectedDataset: semantic["dc:title"],
 			semanticWindowClosed: false,
-			previewLoading: true,
-			semanticJSON: {}
+			previewLoading: false,
+			semanticJSON: semantic
 		});
+	};
 
-		try {
-			let endpoint = `${config.semanticServiceType}/${semantic}`;
-			const response = await fetch(endpoint, {
-				mode: "cors",
-				headers: getHeader(),
-				contentType: "application/json"
-			});
-			const jsonData = await response.json();
-			this.setState({
-				previewLoading: false,
-				semanticJSON: jsonData
-			});
-		} catch (error) {
-			console.error("Error fetching JSON data:", error);
-			this.setState({
-				previewLoading: false
-			});
+	async downloadTemplate() {
+		if (this.state.selectedDataset !== null || this.state.selectedDataset !== undefined) {
+			let semanticName = this.state.selectedDataset;
+			semanticName = semanticName.replace(":", "_")
+			// let filename = `${datasetId}.zip`;
```

Comment:
could you remove this comment

Reply:
yes, I kept that as I wasn't sure if this would work or not because the template API was not available.